Src/main.c

```diff
@@ -192,7 +194,12 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
-
+void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart){
+	if(huart->Instance == huart3.Instance){//input from the PC
```

Comment:
change huart3 to huartx (and change name of huartx to soemthing more usefull, like Huart_Putty). This will make changing huart easier, by only having to change the define.

Reply:
Replaced by PuttyHuart, which is defined in main.c.
````c
#define PuttyHuart huart3 // UART for the communication
`````